### ISSUE TITLE
Refactor losses

### ISSUE THREAD

**Issue by vitusbenson** [earthnet2021/earthnet-models-pytorch#23]
The losses are currently a mess and need a better API.

Goal:
- Move dataset & model specific out of the losses
- More explicit naming
- Docstrings that indicate the maths behind each loss

Requirements:
- Need new abstraction that can handle seperation of loss computation from dataset/model specific stuff
- First version should support:
   - [ ] L1 loss
   - [ ] L2 loss
   - [ ] masking (both by data quality mask & by landcover mask)
   - [ ] Weight decay (as a regularization)

Culprits:
- Where to move the masking of preds & targets? Currently it is computed inside the loss function. However, it might be very dataset specific, so what is a good API for this?

Risks:
- Will likely break backward-compatibility
